File: src/leetcode/array/kotlin/ContainerWithMostWater.kt
package leetcode.array.kotlin


fun maxArea(height: IntArray): Int {
    var res = 0
    var l = 0
    var r = height.size - 1
    while (l < r) {
        var minHeight = if (height[l] < height[r]) height[l++] else height[r--]
        res = Math.max(res, minHeight * (r - l + 1))
    }
    return res
}

File: src/leetcode/array/js/ContainerWithMostWater.js
var maxArea = function(height) {
    let res = 0;
    let l = 0;
    let r = height.length - 1;
    while (l < r) {
        let minHeight = height[l] < height[r] ? height[l++] : height[r--];
        res = Math.max(res, minHeight*(r - l + 1));
    }
    return res;
};